common_features.mk

```diff
@@ -429,3 +429,54 @@ ifeq ($(strip $(DIP_SWITCH_ENABLE)), yes)
   SRC += $(QUANTUM_DIR)/dip_switch.c
   OPT_DEFS += -DDIP_SWITCH_ENABLE
 endif
+
+VALID_EEPROM_DRIVER_TYPES := vendor custom transient i2c
+EEPROM_DRIVER ?= vendor
+ifeq ($(filter $(EEPROM_DRIVER),$(VALID_EEPROM_DRIVER_TYPES)),)
+  $(error EEPROM_DRIVER="$(EEPROM_DRIVER)" is not a valid EEPROM driver)
+else
+  OPT_DEFS += -DEEPROM_ENABLE
+  ifeq ($(strip $(EEPROM_DRIVER)), custom)
+    OPT_DEFS += -DEEPROM_DRIVER -DEEPROM_CUSTOM
+    COMMON_VPATH += $(DRIVER_PATH)/eeprom
```

Comment:
Given this is the custom pathway, do we need to set the vpath? Would the implementation not exist at the keyboard level?

Reply:
`eeprom_driver.h` is basically the frontend for all of the implementations except for the vendor-provided one.
`eeprom_driver.c` provides common implementations for byte/word/dword such that the actual drivers only require implementation of init/clear/readblock/writeblock functions. So, it'd be expected that even if there's a keyboard implementation, they'd still be including the header and only providing the missing methods, much like the template file provided.

Reply:
Yep, this is still required -- `eeconfig.c` still looks for `eeprom_driver.h`.

Reply:
Ignore the errant "resolved" -- hit the wrong button.

Reply:
Attempted to implement a custom eeprom driver in the keyboard directory, fails if VPATH isn't updated, due to only specifying the handful of required functions.

Confirmed that it's still required when using the custom template.